test/modifiers/batch.test.js

```diff
@@ -0,0 +1,284 @@
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(actual).to.deep.equal([[1], [2, 3], [4, 5]]);
+
+          done();
+        });
+
+      input.write(1);
+
+      setTimeout(function () {
+        input.write(2);
+        input.write(3);
+      }, 1);
+
+      setTimeout(function () {
+        input.write(4);
+        input.write(5);
+
+        input.end();
+      }, 6);
+    });
+  });
+
+  describe('when "count" is defined in options', function () {
+    var expected = [[buff('item1'), buff('item2')], [buff('item3'), buff('item4')]];
+    var tempObjData = [].concat(objData);
+    var objExpected = [];
+
+    while (tempObjData.length) {
+      objExpected.push(tempObjData.splice(0, 2));
+    }
+
+    basicTests({ count: 2 }, expected, objExpected);
+
+    it('emits chunks of the defined size', function (done) {
+      var COUNT = 3;
+      var CHUNKS = 5;
+
+      var input = getReadableStream(_.times(CHUNKS * COUNT), {
+        objectMode: true
+      });
+
+      var chunks = 0;
+
+      input
+        .pipe(batch({ count: COUNT }))
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(COUNT);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+    });
+  });
+
+  describe('when both "time" and "count" are defined in options', function () {
+    function asyncWrite(stream, dataArr) {
+      if (dataArr.length) {
+        setTimeout(function () {
+          stream.push(dataArr[0]);
+
+          asyncWrite(stream, dataArr.slice(1));
+        }, 1);
+      } else {
+        stream.end();
+      }
+    }
+
+    it('writes chunks if count is met first before the defined time', function (done) {
+      var input = through.obj();
+      var chunks = 0;
+      var CHUNKS = 10;
+
+      input
+        .pipe(batch({
+          count: 1,
+          time: 5000
+        }))
+        .on('data', function (chunk) {
+          chunks += 1;
+          expect(chunk).to.be.an('array').and.to.have.lengthOf(1);
+        })
+        .on('error', done)
+        .on('end', function () {
+          expect(chunks).to.equal(CHUNKS);
+          done();
+        });
+
+      asyncWrite(input, _.times(CHUNKS));
+    });
+
+    it('writes chunks if time is met first before the defined count', function (done) {
+      var input = through.obj();
+      var CHUNKS = 10;
+      var reads = 0;
+
+      input
+        .pipe(batch({
+          count: 100,
+          time: 1
+        }))
+        .on('data', function (chunk) {
+          reads += 1;
+          expect(chunk).to.be.an('array').and.to.have.length.above(0);
+        })
+        .on('error', done)
+        .on('end', function () {
+          // for good measure, leave some wiggle room as to
+          // how many writes we expect, because timers
+          expect(reads).to.be.above(CHUNKS / 2);
```

Comment:
Done. I made the entire `asyncWrite` helper use fake time, so now all of the tests that use it are predictable.